### DISCUSSION TITLE
Trying to migrate a sandbox from v8 to v9 with no success

### DISCUSSION THREAD

**Discussion by lucax88x** [pmndrs/react-spring#1819] [Support] [answered]
I found this simple example that suits my needs: https://codesandbox.io/s/18y0pky3z7.

I tried to upgrade to v9, but yeah, I cannot make it work, not at all.

This is my forked sandbox with v9: https://codesandbox.io/s/react-spring-tabs-with-direction-forked-rxm2e

I just cannot make animation work.

Thanks if anyone is willing to help me!

**Comment by joshuaellis** [comment, marked as answer]
Hey @lucax88x,

I would probably avoid using state to save the previous tab, why not use a Ref? 

See this sandbox: https://codesandbox.io/s/react-spring-tabs-with-direction-forked-w9up3?file=/src/Tabs.tsx

**Comment by lucax88x** [comment]
@joshuaellis  you da man, works much better.

If you go fast, it's a bit bugged still.

Do you have any better example where I can gather around? Maybe using useSprings?

**Reply by joshuaellis** [reply]
I guess in a way, you're trying to make an infinite carousel, why not check out this blog post? https://ironeko.com/posts/ininite-carousel-with-react-spring-how-to
